File: Leetcode_Python/min_stack.py
class MinStack:
    # @param x, an integer
    # @return an integer
    def push(self, x):
        self.stk.append(x)
        if x <= self.min_list[-1]:
            self.min_list.append(x)
        
    # @return nothing
    def pop(self):
        if self.stk[-1] == self.min_list[-1]:
            del self.min_list[-1]
        del self.stk[-1]

    # @return an integer
    def top(self):
        return self.stk[-1]

    # @return an integer
    def getMin(self):
        return self.min_list[-1]
    
    def __init__(self):
        self.stk = []
        self.min_list = [float("inf")]

File: cleanCode50/MinStack.java
public class MinStack {
    private Stack<Integer> stack;
    private Stack<Integer> minList;

    MinStack() {
        this.stack = new Stack<>();
        this.minList = new Stack<>();
        minList.push(Integer.MAX_VALUE);
    }
    
    public void push(int x) {
        stack.push(x);
        if (x <= minList.peek()) {
            minList.push(x);
        }
    }

    public void pop() {
        if (stack.peek().equals(minList.peek())) {
            minList.pop();
        }
        stack.pop();
    }

    public int top() {
        return stack.peek();
    }

    public int getMin() {
        return minList.peek();
    }
}
